test/functional/interface_rest.py

```diff
@@ -420,6 +420,11 @@ def run_test(self):
 
         deployment_info = self.nodes[0].getdeploymentinfo()
         assert_equal(deployment_info, self.test_rest_request('/deploymentinfo'))
+        assert_equal(deployment_info, self.test_rest_request(f"/deploymentinfo/{bb_hash}"))
+
+        self.generate(self.wallet, 10, sync_fun=self.no_op)
+        deployment_info = self.nodes[0].getdeploymentinfo(bb_hash)
+        assert_equal(deployment_info, self.test_rest_request(f"/deploymentinfo/{bb_hash}"))
```

Comment:
I did it to ensure it is fetching data for old blocks correctly.

Reply:
Yeah I get that, but I don't see why we wouldn't just run the hash-specified test immediately on the older hash to avoid doing this twice? Also just querying the previous hash seems more efficient than running generate?

Upon further thought, I think it does indeed make sense to include it in this PR.

```diff
diff --git a/test/functional/interface_rest.py b/test/functional/interface_rest.py
index 559249101..1ba8f60d9 100755
--- a/test/functional/interface_rest.py
+++ b/test/functional/interface_rest.py
@@ -420,11 +420,10 @@ class RESTTest (BitcoinTestFramework):
 
         deployment_info = self.nodes[0].getdeploymentinfo()
         assert_equal(deployment_info, self.test_rest_request('/deploymentinfo'))
-        assert_equal(deployment_info, self.test_rest_request(f"/deploymentinfo/{bb_hash}"))
 
-        self.generate(self.wallet, 10, sync_fun=self.no_op)
-        deployment_info = self.nodes[0].getdeploymentinfo(bb_hash)
-        assert_equal(deployment_info, self.test_rest_request(f"/deploymentinfo/{bb_hash}"))
+        previous_bb_hash = self.nodes[0].getblockhash(self.nodes[0].getblockcount() - 1)
+        deployment_info = self.nodes[0].getdeploymentinfo(previous_bb_hash)
+        assert_equal(deployment_info, self.test_rest_request(f"/deploymentinfo/{previous_bb_hash}"))
 
         non_existing_blockhash = '42759cde25462784395a337460bde75f58e73d3f08bd31fdc3507cbac856a2c4'
         resp = self.test_rest_request(f'/deploymentinfo/{non_existing_blockhash}', ret_type=RetType.OBJ, status=400)

```

Reply:
> Also just querying the previous hash seems more efficient than running generate?

For sure, and for what we'd like to test may be enough, I'm gonna change it.